Description: Support running grub-probe in grub-legacy's update-grub
Author: Robert Millan <rmh@aybabtu.com>
Author: Colin Watson <cjwatson@debian.org>
Forwarded: not-needed
Last-Update: 2013-12-25

Index: b/util/getroot.c
===================================================================
--- a/util/getroot.c
+++ b/util/getroot.c
@@ -244,6 +244,20 @@
 
   if (ctx->start == part_start)
     {
+      /* This is dreadfully hardcoded, but there's a limit to what GRUB
+         Legacy was able to deal with anyway.  */
+      if (getenv ("GRUB_LEGACY_0_BASED_PARTITIONS"))
+	{
+	  if (partition->parent)
+	    /* Probably a BSD slice.  */
+	    ctx->partname = xasprintf ("%d,%d", partition->parent->number,
+				       partition->number + 1);
+	  else
+	    ctx->partname = xasprintf ("%d", partition->number);
+
+	  return 1;
+	}
+
       ctx->partname = grub_partition_get_name (partition);
       return 1;
     }
